software_training_assignment/src/testing.cpp

```diff
@@ -0,0 +1,63 @@
+#include <ros/ros.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/client/terminal_state.h>
+#include <software_training_assignment/GotoAction.h>
+#include <std_srvs/Empty.h>
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "test_node");
+
+    if (argc != 3)
+    {
+        ROS_INFO("rosrun software_training_assignment testing x y");
+        return 0;
+    }
+
+    ros::NodeHandle n;
+
+    actionlib::SimpleActionClient<software_training_assignment::GotoAction> ac ("turtlesim1/goto", true);
+    
+    ROS_INFO("Waiting for action server to start.");
+
+    ac.waitForServer();
+
+    ROS_INFO("Action server started, sending goal.");
+    
+
+    software_training_assignment::GotoGoal goal;
+
+    goal.x = atoll(argv[1]);
```

Comment:
we don't use command line arguments for nodes usually. most of the time the data is passed through subscriber or publisher callbacks